File: online_judge_solutions/text-analysis.cpp
//text analysis

#include <bits/stdc++.h>

using namespace std;

int main() {
	int n;
	scanf("%d", &n);
	string word;
	cin >> word;

	bool inside = false;
	int biggestWord = 0, qtdInside = 0, nowLen = 0;
	for (int i = 0; i < n; i++) {
		if (word[i] == '(') {inside = true;
			if (nowLen > biggestWord) biggestWord = nowLen;
			nowLen = 0;
			continue;
		}
		else if (word[i] == ')') {inside = false; continue;}

		if (inside) {
			if ((word[i] != '_' && word[i - 1] == '_') || (word[i - 1] == '(' && word[i] != '_'))
				qtdInside++;
		} else {
			if (word[i] == '_') {
				if (nowLen > biggestWord) biggestWord = nowLen;
				nowLen = 0;
				continue;
			} else nowLen++;
		}

		if (nowLen > biggestWord) biggestWord = nowLen;
	}

	cout << biggestWord << " " << qtdInside << endl;

	return 0;
}

File: online_judge_solutions/text-analysis.py
#text analysis

n = input().split("_")

biggestOut = 0
qtdInside = 0

print (n)

if n == "": print ("0 0")
else:
	inside = False
	for i in n:
		if i == "": continue
		
		if i[0] == "(":
			inside = True

		elif i[-1] == ")":
			qtdInside += 1
			inside = False

		if not inside:
			if len(i[0]) > biggestOut:
				biggestOut = len(i[0])

		else:
			qtdInside += 1

print ("%d %d", (biggestOut, qtdInside))
